src/c20_server/flask_server.py

```diff
@@ -45,15 +43,38 @@ def _report_failure():
         user = User(user_id)
         job_manager.report_failure(user)
         return {}, 200
+
     return app
 
 
-if __name__ == '__main__':
+def update_job_manager(job_manager, client_data):
```

Comment:
I like the movement of adding jobs to the job manager into a separate function. It looks better this way.